semgrep-core/src/analyzing/Naming_AST.ml

```diff
@@ -582,7 +582,18 @@ let resolve2 lang prog =
           | ImportAs (_, FileName (s, tok), Some (alias, id_info)) ->
               (* for Go *)
               let sid = H.gensym () in
-              let base = (Filename.basename s, tok) in
+              let pkgname =
+                let pkgpath, pkgbase = Common2.dirs_and_base_of_file s in
+                if pkgbase =~ "v[0-9]+" then
```

Comment:
when there's a long list of if elseif, I actually now tend to prefer
match () with
| _ when pkgbase =~ "v[0-9]+" -> ...
| _ when pkgbase =~ ... -> ...
| _ ->